jdaviz/core/helpers.py

```diff
@@ -35,3 +39,197 @@ def __init__(self, app=None):
+        models : dict
+            A dict of models, with the key being the label name and the value
+            being an `astropy.modeling.CompoundModel` object. Defaults to
+            `self.fitted_models` if no parameter is provided.
+        model_label : str
+            The name of the model that will be found and returned. If it
+            equals default, every model present will be returned.
+        x : int
+            The x coordinate of the model spaxels that will be returned.
+        y : int
+            The y coordinate of the model spaxels that will be returned.
+
+        Returns
+        -------
+        :dict: dictionary of the selected models.
+        """
+        selected_models = {}
+        # If models is not provided, use the app's fitted models
+        if not models:
+            models = self.fitted_models
+
+        # Loop through all keys in the dict models
+        for label in models:
+            # If no label was provided, use label name without coordinates.
+            if model_label == "default" and " (" in label:
+                find_label = label.split(" (")[0]
+            # If coordinates are not present, just use the label.
+            elif model_label == "default":
+                find_label = label
+            else:
+                find_label = model_label
+
+            # If x and y is set, return keys that match the model plus that
+            # coordinate pair. If only x or y is set, return keys that fit
+            # that value for the appropriate coordinate.
+            if x is not None and y is not None:
+                find_label = r"{} \({}, {}\)".format(find_label, x, y)
+            elif x:
+                find_label = r"{} \({}, .+\)".format(find_label, x)
+            elif y:
+                find_label = r"{} \(.+, {}\)".format(find_label, y)
+
+            if re.search(find_label, label):
+                selected_models[label] = models[label]
+
+        return selected_models
+
+    def get_model_parameters(self, models=None, model_label=None):
+        """
+        Convert each parameter of model inside models into a coordinate that
+        maps the model name and parameter name to a `astropy.units.Quantity`
+        object.
+
+        Parameters
+        ----------
+        models : dict
+            A dictionary where the key is a model name and the value is an
+            `astropy.modeling.CompoundModel` object.
+        model_label : str
+            Get model parameters for a particular model by inputting its label.
+
+        Returns
+        -------
+        :dict: a dictionary of the form
+            {model name: {parameter name: [[`astropy.units.Quantity`]]}}
+            for 3d models or
+            {model name: {parameter name: `astropy.units.Quantity`}} where the
+            Quantity object represents the parameter value and unit of one of
+            spaxel models or the 1d models, respectively.
+        """
+        if not models and model_label:
```

Comment:
It's probably worth covering the case where both `model` and `model_label` were input, since it's possible.

Reply:
Good call, done.